mm2src/mm2_main/src/lp_swap/swap_watcher.rs

```diff
@@ -513,14 +514,12 @@ impl LastState for Stopped {
     async fn on_changed(self: Box<Self>, _watcher_ctx: &mut Self::Ctx) -> Self::Result {}
 }
 
-pub async fn process_watcher_msg(ctx: MmArc, msg: &[u8]) {
+pub fn process_watcher_msg(ctx: MmArc, msg: &[u8]) -> P2PRequestResult<()> {
     let msg = match decode_signed::<SwapWatcherMsg>(msg) {
         Ok(m) => m,
-        Err(watcher_msg_err) => {
-            error!("Couldn't deserialize 'SwapWatcherMsg': {:?}", watcher_msg_err);
-            // Drop it to avoid dead_code warning
-            drop(watcher_msg_err);
-            return;
+        Err(e) => {
+            error!("Couldn't deserialize 'SwapWatcherMsg': {:?}", e);
+            return MmError::err(P2PRequestError::DecodeError(e.to_string()));
```

Comment:
The logic of the logging and error processing is not complete

Reply:
I didn't quite understand

Reply:
That is the place where `process_watcher_msg`.

```rust
if ctx.is_watcher() && lp_swap::process_watcher_msg(ctx.clone(), &message.data).is_err() {
    return;
}
```                

As you can see `MmError` is not utilized. No use to allocate memory for string and return this type of error, the incident is already processed.

Reply:
https://github.com/KomodoPlatform/atomicDEX-API/pull/1755#discussion_r1168395931